src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline

```diff
@@ -3416,11 +3418,21 @@ public class TuringProgCallableStreamingBidi {
           .hasArg(true)
           .longOpt("image_file_name")
           .build());
+    options.addOption(
+        Option.builder("")
+          .required(false)
+          .hasArg(true)
+          .longOpt("stage")
+          .build());
 
     CommandLine cl = (new DefaultParser()).parse(options, args);
     String imageFileName = cl.getOptionValue("image_file_name", "image_file.jpg");
+    Comment.Stage stage =
+      cl.getOptionValue("stage") != null
+          ? Comment.Stage.valueOf(cl.getOptionValue("stage"))
```

Comment:
What happens if a non-enum value is provided on the command line? Does this blow up?

In general, how should we handle this across languages? We should make an explicit decision.

The most user-friendly, ideal approach would be to check against allowed values and error if the value provided is not one of those (and one hopes most languages have idiomatic ways of doing this check) with a helpful message printing out a list of allowed values.

We may decide this is too much work for samplegen and defer to each language to blow up as it sees fit. I like this less but it may be the pragmatic initial approach.

We should be explicit about which approach we choose and document that in the official feature spec sheet.

What are your thoughts, @hzyi-google @beccasaurus ?

Reply:
The way samples translate enum literals to enum values is pretty much the language-idiomatic way to get an enum value from its string representation. So I think the language features will provide users with enough information should they pass an invalid parameter.

For example, the `valueOf` method in Java will throw an `IllegalArgumentException` with an error message showing the value given cannot be associated with an enum.

Ruby and Node takes a symbol/string directory for enums, so nothing needs to be done.

For Python/PHP, we are using the helper methods provided by protoc/protobuf runtime.

Reply:
Sure, I get that we're doing things in the idiomatic way, but I wonder whether for a sample it would be more helpful, instead of raising an exception "this is an invalid value", to show an informative error "the value must be one of FOO, BAR, BAZ". I assume the language constructs don't (all?) do this.

The more I think about it, the more I think this would be both (a) desirable for us to do eventually and (b) not blocking on us right now. We can enhance this later.